frontends/web/src/routes/exchange/bitrefill.tsx

```diff
@@ -0,0 +1,235 @@
+ */
+
+import { useState, useEffect, useRef, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Header } from '@/components/layout';
+import { Spinner } from '@/components/spinner/Spinner';
+import { AccountCode, IAccount, proposeTx, sendTx, TTxInput } from '@/api/account';
+import { findAccount, isBitcoinOnly } from '@/routes/account/utils';
+import { useDarkmode } from '@/hooks/darkmode';
+import { getConfig } from '@/utils/config';
+import style from './iframe.module.css';
+import { i18n } from '@/i18n/i18n';
+import { alertUser } from '@/components/alert/Alert';
+import { parseExternalBtcAmount } from '@/api/coins';
+import { useLoad } from '@/hooks/api';
+import { BitrefillTerms, localeMapping } from '@/components/terms/bitrefill-terms';
+import { getBitrefillInfo } from '@/api/exchanges';
+import { getURLOrigin } from '@/utils/url';
+
+// Map coins supported by Bitrefill
+const coinMapping: Readonly<Record<string, string>> = {
+  btc: 'bitcoin',
+  tbtc: 'bitcoin',
+  ltc: 'litecoin',
+  eth: 'ethereum',
+  usdt: 'usdt_erc20',
+  usdc: 'usdc_erc20',
+};
+
+type TProps = {
+  accounts: IAccount[];
+  code: AccountCode;
+};
+
+export const Bitrefill = ({ accounts, code }: TProps) => {
+  const { t } = useTranslation();
+  const { isDarkMode } = useDarkmode();
+  const account = findAccount(accounts, code);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const iframeRef = useRef<HTMLIFrameElement | null>(null);
+  const [iframeLoaded, setIframeLoaded] = useState(false);
+  const [height, setHeight] = useState(0);
+  const resizeTimerID = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const bitrefillInfo = useLoad(() => getBitrefillInfo('spend', code));
+
+  const config = useLoad(getConfig);
+  const [agreedTerms, setAgreedTerms] = useState(false);
+
+  const hasOnlyBTCAccounts = accounts.every(({ coinCode }) => isBitcoinOnly(coinCode));
+
+  useEffect(() => {
+    if (config) {
+      setAgreedTerms(config.frontend.skipBitrefillWidgetDisclaimer);
+    }
+  }, [config]);
+
+  const onResize = useCallback(() => {
+    if (resizeTimerID.current) {
+      clearTimeout(resizeTimerID.current);
+    }
+    resizeTimerID.current = setTimeout(() => {
+      if (containerRef.current) {
+        setHeight(containerRef.current.offsetHeight);
+      }
+    }, 200);
+  }, []);
+
+  useEffect(() => {
+    onResize();
+    window.addEventListener('resize', onResize);
+    return () => {
+      window.removeEventListener('resize', onResize);
+      if (resizeTimerID.current) {
+        clearTimeout(resizeTimerID.current);
+      }
+    };
+  }, [onResize]);
+
+  const handleMessage = useCallback(async (event: MessageEvent) => {
+    if (
+      !account
+      || !bitrefillInfo?.success
+      || ![getURLOrigin(bitrefillInfo.url), 'https://embed.bitrefill.com'].includes(event.origin)
+    ) {
+      return;
+    }
+
+    const data = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
```

Comment:
JSON.parse could throw an error if `event.data` is not valid JSON, this probably never happens, however it is coming from outside the BitBoxApp so it would not be wrong to wrap in try/catch